geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionAlterFunction.java

```diff
@@ -237,7 +237,7 @@ public void execute(FunctionContext context) {
           }
         }
 
-        if (!nameFound) {
+        if (!nameFound && !newCacheListenerName.isEmpty()) {
```

Comment:
instead of adding the check here, would it be better to add the check in line after 232:
if(newCacheListnerName.isEmpty()) break;